src/datetime/mod.rs

```diff
@@ -282,7 +282,8 @@ impl<Tz: TimeZone> DateTime<Tz> {
             subsec_nanos -= 1_000_000_000;
             timestamp += 1;
         }
-        try_opt!(timestamp.checked_mul(1_000_000_000)).checked_add(subsec_nanos)
+        let ts = try_err!(ok_or!(timestamp.checked_mul(1_000_000_000), Error::OutOfRange));
```

Comment:
If I remember correctly, we want the value to be unwrapped anyway in all instances. I sugges `try_ok_or` instead that directly unwraps T or returns the error early.

`try_err!(ok_or!(..))` vs `try_ok_or!`

I don't really mind chaining macros, but wanted to have it mentioned.